src/controller/event/handler/webhook/scan/delete.go

```diff
@@ -48,32 +48,20 @@ func (o *DelArtHandler) Handle(value interface{}) error {
 
 	log.Debugf("clear the scan reports as receiving event %s", evt.EventType)
 
-	digests := make([]string, 0)
-	query := &q.Query{
-		Keywords: make(map[string]interface{}),
-	}
-
-	ctx := orm.NewContext(context.TODO(), bo.NewOrm())
+	ctx := o.Context()
 	// Check if it is safe to delete the reports.
-	query.Keywords["digest"] = evt.Artifact.Digest
-	l, err := artifact.Ctl.List(ctx, query, nil)
-
-	if err != nil && len(l) != 0 {
+	count, err := artifact.Ctl.Count(ctx, q.New(q.KeyWords{"digest": evt.Artifact.Digest}))
+	if err != nil {
 		// Just logged
 		log.Error(errors.Wrap(err, "delete image event handler"))
-		// Passed for safe consideration
-	} else {
-		if len(l) == 0 {
-			digests = append(digests, evt.Artifact.Digest)
-			log.Debugf("prepare to remove the scan report linked with artifact: %s", evt.Artifact.Digest)
+	} else if count == 0 {
+		log.Debugf("prepare to remove the scan report linked with artifact: %s", evt.Artifact.Digest)
 
+		if err := scan.DefaultController.DeleteReports(ctx, evt.Artifact.Digest); err != nil {
+			return errors.Wrap(err, "delete image event handler")
```

Comment:
delete aritfact

Reply:
When the artifact was deleted, the report for the digest of the artifact will be deleted when the digest does not exist in the artifact table.